hamilton/cli/__main__.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from pathlib import Path
+from pprint import pprint
+from typing import Annotated, Any, Optional
+
+import typer
+from pydantic import BaseModel
+
+from hamilton import driver
+from hamilton.cli import commands
+
+logger = logging.getLogger(__name__)
+
+
+class Response(BaseModel):
+    command: str
+    success: bool
+    message: Any
+
+
+class CliState:
+    verbose: Optional[bool] = None
+    json_out: Optional[bool] = (None,)
+    dr: Optional[driver.Driver] = None
+    dataflow_version: Optional[dict] = None
+
+
+cli = typer.Typer()
+state = CliState()
+
+
+# entrypoint for `hamilton` without command
+@cli.callback()
+def main(
+    verbose: Annotated[bool, typer.Option(help="Output all intermediary commands")] = False,
+    json_out: Annotated[
+        bool, typer.Option(help="Output JSON for programmatic use (e.g., CI)")
+    ] = False,
+):
+    """Hamilton CLI"""
+    state.verbose = verbose
+    state.json_out = json_out
+
+
+@cli.command()
+def build(
+    ctx: typer.Context,
+    modules: Annotated[
+        list[Path], typer.Argument(exists=True, dir_okay=False, readable=True, resolve_path=True)
+    ],
+):
+    """Build dataflow with MODULES"""
+    try:
+        config = dict()
```

Comment:
Add a TODO for the config loading -- good to pass in a json file but not needed now